### PR TITLE
[BEAM-1644] Move travis & jenkins into shared test-infra dir, and move k8s scripts there

### PR DESCRIPTION
**Pull request by ssisk**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [X] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [X] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [X] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [X] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
Reasoning is discussed in the bug, but generally, we want to have a language neutral place for these kubernetes scripts to live, and also don't want a million top level directories for test-infra

cc @jasonkuster

### CODE REVIEW

**Comment by jasonkuster** [comment]
LGTM

File: .travis.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ addons:
     - python2.7
 env:
   global:
-   - MAVEN_OVERRIDE="--settings=.travis/settings.xml"
+   - MAVEN_OVERRIDE="--settings=.test-infra/travis/settings.xml"
    - MAVEN_CONTAINER_OVERRIDE="-DbeamSurefireArgline='-Xmx512m'"
```
Comment (ssisk) [inline .travis.yml]: jason pointed out there's another .travis references in this file

Comment (ssisk) [inline .travis.yml]: This is fixed @jasonkuster please take a look (again)

**Comment by ssisk** [comment]
retest this please

**Comment by ssisk** [comment]
retest this please

**Comment by ssisk** [comment]
Test failures so far: 
------------
https://builds.apache.org/job/beam_PreCommit_Java_MavenInstall/9376/
Failed Tests: 1
org.apache.beam.runners.apex.translation.ParDoTranslatorTest.testAssertionFailure

failure details: https://builds.apache.org/job/beam_PreCommit_Java_MavenInstall/9376/org.apache.beam$beam-runners-apex/testReport/org.apache.beam.runners.apex.translation/ParDoTranslatorTest/testAssertionFailure/

java.lang.AssertionError: assertion should have failed
	at org.junit.Assert.fail(Assert.java:88)
	at org.apache.beam.runners.apex.translation.ParDoTranslatorTest.runExpectingAssertionFailure(ParDoTranslatorTest.java:160)
	at org.apache.beam.runners.apex.translation.ParDoTranslatorTest.testAssertionFailure(ParDoTranslatorTest.java:175)

------------
https://builds.apache.org/job/beam_PreCommit_Java_MavenInstall/9381/
Failed Tests: org.apache.beam.runners.spark.translation.streaming.CreateStreamTest.testLateDataAccumulating

error details:
https://builds.apache.org/job/beam_PreCommit_Java_MavenInstall/9381/org.apache.beam$beam-runners-spark/testReport/junit/org.apache.beam.runners.spark.translation.streaming/CreateStreamTest/testLateDataAccumulating/

java.lang.AssertionError: Flatten.Iterables/FlattenIterables/FlatMap/ParMultiDo(Anonymous).out0: 
Expected: iterable over [<1>, <2>, <3>] in any order
     but: Not matched: <4>


so we've seen different errors both times so far

**Comment by ssisk** [comment]
Console output of the jenkins run shows that the build succeeded - 2017-04-12T20:31:56.948 [INFO] ------------------------------------------------------------------------
2017-04-12T20:31:56.948 [INFO] Reactor Summary:
2017-04-12T20:31:56.948 [INFO] 
2017-04-12T20:31:56.948 [INFO] Apache Beam :: Parent .............................. SUCCESS [  8.887 s]
2017-04-12T20:31:56.948 [INFO] Apache Beam :: SDKs :: Java :: Build Tools ......... SUCCESS [ 23.823 s]
2017-04-12T20:31:56.948 [INFO] Apache Beam :: SDKs ................................ SUCCESS [ 47.948 s]
2017-04-12T20:31:56.948 [INFO] Apache Beam :: SDKs :: Common ...................... SUCCESS [ 26.703 s]
2017-04-12T20:31:56.948 [INFO] Apache Beam :: SDKs :: Common :: Fn API ............ SUCCESS [ 57.870 s]
2017-04-12T20:31:56.948 [INFO] Apache Beam :: SDKs :: Common :: Runner API ........ SUCCESS [ 30.498 s]
2017-04-12T20:31:56.948 [INFO] Apache Beam :: SDKs :: Java ........................ SUCCESS [ 11.154 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: Core ................ SUCCESS [03:56 min]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: IO .................. SUCCESS [01:00 min]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: IO :: Common ........ SUCCESS [ 10.742 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: Runners ............................. SUCCESS [ 10.122 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: Runners :: Core Java Construction ... SUCCESS [01:04 min]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: Runners :: Core Java ................ SUCCESS [01:56 min]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: Runners :: Direct Java .............. SUCCESS [03:15 min]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: IO :: Elasticsearch . SUCCESS [01:21 min]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: IO :: Google Cloud Platform SUCCESS [01:26 min]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: IO :: Hadoop Common . SUCCESS [ 21.839 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: IO :: Hadoop ........ SUCCESS [  9.938 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: IO :: Hadoop :: input-format SUCCESS [ 43.659 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: IO :: Kafka ......... SUCCESS [ 27.039 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: Runners :: Spark .................... SUCCESS [03:03 min]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: Runners :: Google Cloud Dataflow .... SUCCESS [ 52.367 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: IO :: Hadoop :: jdk1.8-tests SUCCESS [01:18 min]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: IO :: HBase ......... SUCCESS [01:14 min]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: IO :: HDFS .......... SUCCESS [ 36.628 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: IO :: JDBC .......... SUCCESS [ 36.079 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: IO :: JMS ........... SUCCESS [ 25.569 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: IO :: Kinesis ....... SUCCESS [ 30.759 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: IO :: MongoDB ....... SUCCESS [ 44.615 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: IO :: MQTT .......... SUCCESS [ 26.196 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: Maven Archetypes .... SUCCESS [  2.523 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: Maven Archetypes :: Starter SUCCESS [ 27.338 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: Maven Archetypes :: Examples SUCCESS [ 40.976 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: Maven Archetypes :: Examples - Java 8 SUCCESS [ 26.149 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: Extensions .......... SUCCESS [  5.437 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: Extensions :: Jackson SUCCESS [ 26.110 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: Extensions :: Join library SUCCESS [ 19.041 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: Extensions :: Sorter  SUCCESS [ 39.880 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: Harness ............. SUCCESS [ 43.927 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: Java 8 Tests ........ SUCCESS [ 13.227 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Python ...................... SUCCESS [13:00 min]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: Runners :: Flink .................... SUCCESS [  1.777 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: Runners :: Flink :: Core ............ SUCCESS [ 54.917 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: Runners :: Flink :: Examples ........ SUCCESS [ 20.860 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: Runners :: Apex ..................... SUCCESS [01:16 min]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: Examples ............................ SUCCESS [  1.641 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: Examples :: Java .................... SUCCESS [21:16 min]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: Examples :: Java 8 .................. SUCCESS [ 27.892 s]
2017-04-12T20:31:56.949 [INFO] Apache Beam :: SDKs :: Java :: Aggregated Javadoc .. SUCCESS [ 32.321 s]
2017-04-12T20:31:56.949 [INFO] ------------------------------------------------------------------------
2017-04-12T20:31:56.949 [INFO] BUILD SUCCESS
2017-04-12T20:31:56.949 [INFO] ------------------------------------------------------------------------
2017-04-12T20:31:56.949 [INFO] Total time: 01:26 h
2017-04-12T20:31:56.950 [INFO] Finished at: 2017-04-12T20:31:56+00:00
2017-04-12T20:31:57.619 [INFO] Final Memory: 298M/1852M
2017-04-12T20:31:57.620 [INFO] ------------------------------------------------------------------------
Waiting for Jenkins to finish collecting data
channel stopped

**Comment by davorbonaci** [comment]
Run Seed Job

**Pull request closed without merging**